client-v2/src/redux/modules/pageViewData/actions.ts

```diff
@@ -55,7 +56,12 @@ const getPageViewData = (
     .map(_ => selectArticleFromArticleFragment(state, _))
     .filter(_ => _) as DerivedArticle[];
   const urlPaths: string[] = articles
-    .map(article => article.urlPath)
+    .map(
+      article =>
+        article.urlPath
+          ? article.urlPath // it's an article
+          : article.href && !isValidURL(article.href) && article.href.substr(1) // it's a snaplink
```

Comment:
The `!isValidURL(...)` check here ensures that we don't try to request Ophan data for a non-Guardian link. I'm definitely open to different / better ways of making this distinction.

Reply:
What about `isGuardianUrl` or `isGuardianWebsiteUrl` from https://github.com/guardian/facia-tool/blob/master/client-v2/src/shared/util/url.ts#L42

I guess so it's not a guardian article but it is a guardian link, so we may have data for it.

Reply:
The advantage of `isGuardianWebsiteUrl` is that we won't try to fetch data for commercial pages etc

Reply:
@ajwl I considered that, but I don't think that at this stage we have access to a complete URL containing a host name. The `href` is just something like "/news/series/todayinfocus" or "/uk" which would fail both the `isValidURL` and `isGuardianWebsiteUrl` checks. 

Hopefully this example illustrates my thought process:

![Screenshot 2019-10-04 at 15 51 17](https://user-images.githubusercontent.com/12645938/66217244-d9936100-e6be-11e9-9ed8-cb4bda084e35.png)

The third snap is not part of the Ophan query, because although it _is_ a valid URL, it is not one that Ophan would store data on (because the domain is not just the base Guardian domain). If all commercial pages are located on something like http://some-domain.theguardian.com then we would not, in this implementation, request data on them.

Happy to document this more explicitly in the code, if you think that would help?

Reply:
👍  makes sense. And have just seen it working in local